app/lib/activitypub/activity/create.rb

```diff
@@ -69,10 +70,61 @@ def process_status_params
         conversation: conversation_from_uri(@object['conversation']),
         media_attachment_ids: process_attachments.take(4).map(&:id),
         poll: process_poll,
+        activity_pub_type: @object['type']
       }
     end
   end
 
+  class Handler < ::Ox::Sax
+    attr_reader :srcs
+    attr_reader :alts
+    def initialize(block)
+      @stack = []
+      @srcs = []
+      @alts = {}
+    end
+
+    def start_element(element_name)
+      @stack << [element_name, {}]
+    end
+
+    def end_element(element_name)
+      self_name, self_attributes = @stack[-1]
+      if self_name == :img && !self_attributes[:src].nil?
+        @srcs << self_attributes[:src]
+        @alts[self_attributes[:src]] = self_attributes[:alt]
+      end
+      @stack.pop
+    end
+
+    def attr(attribute_name, attribute_value)
+      _name, attributes = @stack.last
+      attributes[attribute_name] = attribute_value
+    end
+  end
+
+  def process_inline_images
+    proc = Proc.new { |name| puts name }
+    handler = Handler.new(proc)
+    Ox.sax_parse(handler, @object['content'])
+    handler.srcs.each do |src|
+      if skip_download?
+        @object['content'].gsub!(src, '')
+        next
+      end
+
+      media_attachment = MediaAttachment.create(account: @account, remote_url: src, description: handler.alts[src], focus: nil)
+      media_attachment.file_remote_url = src
+      media_attachment.save
+      if unsupported_media_type?(media_attachment.file.content_type)
```

Comment:
This is only the _reported_ content-type from the remote server, but I think checking against this is consistent with how Mastodon does it (I couldn't find any code where mastodon checks to see if received MIME types are consistent with, for example, actual file headers).